plugins/notification/src/index.ts

```diff
@@ -226,6 +232,159 @@ export interface NotificationPreview extends Class<Doc> {
  */
 export type NotificationClientFactoy = () => NotificationClient
 
+/**
+ * @public
+ */
+export interface ActivityMessage extends Doc {
+  attachedTo: Ref<Doc>
+  attachedToClass: Ref<Class<Doc>>
+
+  modifiedBy: Ref<Account>
+  modifiedOn: Timestamp
+
+  reactions?: number
+}
+
+export type DisplayActivityMessage = DisplayDocUpdateMessage | DisplayChatMessage
+
+export interface DisplayDocUpdateMessage extends DocUpdateMessage {
+  previousMessages?: DocUpdateMessage[]
+  combinedMessagesIds?: Ref<DocUpdateMessage>[]
+}
+
+export interface DisplayChatMessage extends ChatMessage {}
+
+export type ActivityMessageExtensionKind = 'action' | 'footer'
+
+/**
+ * @public
+ */
+export interface ActivityMessageExtension extends Doc {
+  ofMessage: Ref<Class<ActivityMessage>>
+  components: { kind: ActivityMessageExtensionKind, component: AnyComponent }[]
+}
+
+/**
+ * @public
+ */
+export interface DocUpdateMessage extends ActivityMessage {
+  objectId: Ref<Doc>
+  objectClass: Ref<Class<Doc>>
+
+  txId: Ref<TxCUD<Doc>>
+
+  action: DocUpdateAction
+  collection?: string
+  attributesChanges: DocUpdateAttributes[]
```

Comment:
When it could be many of them? I supposed it will be one DocUpdateMessage per change.

Reply:
I create one `DocUpdateMessage` per transaction, but a transaction can have multiple attributes changed.
It looks like this is not a common case, and most often we only have one attribute changed.

For example, when we move a issue to another project we have transaction with multiple changes

Reply:
Yes, let's think about have one change per attribute, even in this scenario to make it more simple for UI.

Reply:
Okay, I send `DocUpdateMessage` per attribute change if we have multiple changes.